File: app/src/main/java/ian_ellis/kotlinmvvm/presentation/handlers/SelectedHandler.java
package ian_ellis.kotlinmvvm.presentation.handlers;

import android.support.annotation.NonNull;
import android.view.View;

public class SelectedHandler {
  
  private SelectedListener selectedListener;
  private SelectedListener actionListener;

  public SelectedHandler(SelectedListener selectedListener, SelectedListener actionListener){
    this.selectedListener = selectedListener;
    this.actionListener = actionListener;
  }

  public void selected(View v){
    selectedListener.selected(v);
  }

  public void action(View v){
    actionListener.selected(v);
  }

  public interface SelectedListener {
    void selected(@NonNull View v);
  }


}

File: app/src/main/java/ian_ellis/kotlinmvvm/presentation/handlers/SelectedHandler.kt
package ian_ellis.kotlinmvvm.presentation.handlers

import android.content.DialogInterface
import android.databinding.BindingAdapter
import android.view.View
import ian_ellis.kotlinmvvm.domain.UiToDoListItem


//class SelectedHandler (val item:UiToDoListItem,val editHandler: ((UiToDoListItem) -> Unit)?){
class SelectedHandler{
  companion object{

    @BindingAdapter(value="app:selected")
    @JvmStatic
    public fun onClick(v:View,listener:View.OnClickListener){
      v.setOnClickListener(listener)
    }
  }

  public fun selected(v:View){
    val tmp = ""
//    editHandler?.invoke(item)
  }
}